nomad/node_endpoint.go

```diff
@@ -253,16 +253,42 @@ func (n *Node) Deregister(args *structs.NodeDeregisterRequest, reply *structs.No
 	}
 	defer metrics.MeasureSince([]string{"nomad", "client", "deregister"}, time.Now())
 
-	// Check node permissions
+	// deregister takes a batch
+	repack := &structs.NodeBatchDeregisterRequest{
+		NodeIDs:      []string{args.NodeID},
+		WriteRequest: args.WriteRequest,
+	}
+
+	return n.deregister(func() (interface{}, uint64, error) {
+		return n.srv.raftApply(structs.NodeDeregisterRequestType, args)
+	}, repack, reply)
+}
+
+// BatchDeregister is used to remove client nodes from the cluster.
+func (n *Node) BatchDeregister(args *structs.NodeBatchDeregisterRequest, reply *structs.NodeUpdateResponse) error {
+	if done, err := n.srv.forward("Node.BatchDeregister", args, args, reply); done {
+		return err
+	}
+	defer metrics.MeasureSince([]string{"nomad", "client", "batch_deregister"}, time.Now())
+	return n.deregister(func() (interface{}, uint64, error) {
+		return n.srv.raftApply(structs.NodeBatchDeregisterRequestType, args)
+	}, args, reply)
+}
+
+// deregister takes a raftMessage closure, to support both Deregister and BatchDeregister
+func (n *Node) deregister(raftMessage func() (interface{}, uint64, error),
+	args *structs.NodeBatchDeregisterRequest,
+	reply *structs.NodeUpdateResponse) error {
```

Comment:
I would suggest ordering them in the following `args, reply, raftApplyFn`.  It's a bit more consistent with rest of functions, and it's a bit nicer visually for the lambda function to appear last.  Also, calling `raftApplyFn` or `applyRaftMsgFn` probably captures meaning better.